pkgs/development/python-modules/patsy/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, fetchurl
+, buildPythonPackage
+, nose
+, six
+, numpy
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "patsy";
+  version = "0.4.1";
+
+  src = fetchurl {
+    url = "https://github.com/pydata/patsy/archive/v${version}.zip";
+    sha256 = "1wkjd71rx1akfrk6iimi4zm783gagxmkxm3ppjyr2c98asni009c";
+  };
+
+  buildInputs = [ nose ];
+  checkPhase = "nosetests -v";
+
+  propagatedBuildInputs = [
+    six
+    numpy
+  ] ++ lib.optional (!isNull scipy) [ scipy ];
```

Comment:
that lib.optional needs to be removed

Reply:
Why? It's an optional dependency to patsy allowing for some optional features.

Reply:
Its no problem if one would pass in `scipy = null;`;

Reply:
Oh, I see! Of course then.